com.hudren.woodpile/src/com/hudren/woodpile/model/LogEvent.java

```diff
@@ -242,21 +244,25 @@ public LogEvent( final String host, final ILoggingEvent event )
 			lineNumber = null;
 		}
 
-		String[] rep = null;
+		int depth = 0;
+		List<String> rep = new ArrayList<>();
 		IThrowableProxy thrown = event.getThrowableProxy();
-		if ( thrown != null )
+		while ( thrown != null )
 		{
 			StackTraceElementProxy[] stack = thrown.getStackTraceElementProxyArray();
-			rep = new String[ stack.length + 1 ];
 
-			int i = 1;
-			rep[ 0 ] = thrown.getClassName() + ": " + thrown.getMessage();
+			String causedBy = depth > 0 ? "Caused by: " : "";
+			rep.add( causedBy + thrown.getClassName() + ": " + thrown.getMessage() );
+
 			for ( StackTraceElementProxy element : stack )
 			{
```

Comment:
Unnecessary curly braces are generally discouraged.

Reply:
I imported the Eclipse config. Is it not configured to remove them?

Reply:
No, last time I updated the config, it was too aggressive in removing them. I like them when there's something nested that requires them. Like when there are two for loops and the inner loop requires braces, then I add the braces to the outer loop.